core/vm/evm.go

```diff
@@ -139,7 +139,7 @@ func NewEVM(ctx Context, statedb StateDB, chainConfig *params.ChainConfig, vmCon
 		interpreters: make([]Interpreter, 1),
 	}
 
-	evm.interpreters[0] = NewEVMInterpreter(evm, vmConfig)
+	evm.interpreters[0] = NewEVMC(evm, vmConfig)
```

Comment:
Since the merge of #17093, you can support other interpreters; this disables the evm interpreter which is not acceptable. What should happen is that if `--evmc-vm` is passed as an argument, then an EVMC interpreter should be inserted before. Something like:

```golang
if cfg.EVMCPath != nil {
  old := evm.interpreters
  evm.interpreters = make([]Interpreter, len(evm.interpreters)+1)
  evm.interpreters[0] = NewEVMC(evm, vmConfig)
  copy(evm.interpreters[1:], old)
}
```